### ISSUE TITLE
docs need updating for usage around `only_tag`

### ISSUE THREAD

**Issue by ghostsquad** [concourse/git-resource#204]
TLDR; the code appears to be **correct**, the docs appear to not clarify the changed behavior when `only_tag` is provided. `HEAD` is not used when `only_tag` is `true`.

---

I was a little concerned that I'd be unable to do the following because `tag` says it tags the `head` of a branch, which is not what I actually want to tag if I'm not making any other changes. But after looking at the code, the code appears to do exactly what I want, so I think the docs need clarification. Can someone sanity check me here?

To further clarify why `HEAD` isn't what should be tagged, HEAD != `get: source, passed: [first]` if the first job fails. So when the second job is run, I want to make sure it's only tagging the revision that passed the build.

```yaml
jobs:
- name: first
  steps:
  - get: source
  - task: <build>
- name: second
  steps:
  - get: source
    passed: [first]
  - put: source
    params:
      tag: foo
      only_tag: true
```

**Comment by taylorsilva** [comment]
> So when the second job is run, I want to make sure it's only tagging the revision that passed the build.

Concourse guarantees this, the resource has no control over this. Unclear how the README for this resource needs to change.
